test/array_family/test_identifiers_handling.py

```diff
@@ -0,0 +1,105 @@
+"""Test for new experiment identifier features"""
+from dials.array_family import flex
+from dxtbx.model import Experiment, ExperimentList
+
+
+def test_selection_identifier_propagation():
+    """
+    If experiment idenfitiers are set, then when selecting on a reflection
+    table, copy across any for the same id.
+
+    If no identifiers are set, then nothing should happen."""
+
+    # Test when identifiers & id column set.
+    refl = flex.reflection_table()
+    refl["id"] = flex.int([0, 0, 0, 1, 1, 1, 2, 2, 2])
+    refl.experiment_identifiers()[0] = "0"
+    refl.experiment_identifiers()[1] = "1"
+    refl.experiment_identifiers()[2] = "2"
+
+    new_refl = refl.select(refl["id"] == 0)
```

Comment:
Might be worth testing using `refl["id"] == 1` instead as a demonstration that identifiers are _not_ reset to start from 0 in the new table